### PR TITLE
Cleanup explicit `EmbeddedResource` items in `System.Windows.Forms.Design.csproj`

### PR DESCRIPTION
**Pull request by elachlan**
For `System.Windows.Forms.Design` this uses globbing to add the Embedded Resources instead of explicit items.

If this goes okay, I will look into the rest of the projects in the solution.

Related: #49

Reference:
https://stackoverflow.com/questions/46584499/how-to-add-a-glob-for-resx-files-for-new-sdk-csproj-file

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8358)

### CODE REVIEW

**Review by lonitra — APPROVED** [review]
This way of globbing LGTM. Thanks for doing this!

File: src/System.Windows.Forms.Design/src/System.Windows.Forms.Design.csproj [hunk 1]
```diff
@@ -72,146 +72,23 @@
   <ItemGroup>
     <Compile Include="..\..\Common\src\RTLAwareMessageBox.cs" Link="Common\RTLAwareMessageBox.cs" />
   </ItemGroup>
+  
+  <!-- Embedded Resources are globbed based on their location. -->
   <ItemGroup>
-    <EmbeddedResource Include="Resources\System\ComponentModel\Design\Arrow.ico">
-      <LogicalName>System.ComponentModel.Design.Arrow</LogicalName>
+    <EmbeddedResource Include="Resources\System\ComponentModel\Design\*.ico">
+      <LogicalName>$System.ComponentModel.Design.([System.String]::Copy('%(FileName)')).ico</LogicalName>
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System.Windows.Forms.Design.csproj:79]: @elachlan, What is the logical name we are ending up here? for ex: Arrow.ico?

**Comment by dreddy-work** [comment]
Also, we need unit tests around these.

File: src/System.Windows.Forms.Design/src/System.Windows.Forms.Design.csproj (hunk 1 above)
Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System.Windows.Forms.Design.csproj:79]: Great catch! it shouldn't have .ico in the logical name. I'll update soon.

**Comment by elachlan** [comment]
@dreddy-work I think that sorts it. I should be able to do the same in `System.Windows.Forms` pretty easily now. I can do that in another PR.

**Comment by dreddy-work** [comment]
> @dreddy-work I think that sorts it. I should be able to do the same in `System.Windows.Forms` pretty easily now. I can do that in another PR.

THanks @elachlan. Can you come up with some unit tests here? This PR showed we have zero coverage around embedded resources (PR succeeded with incorrect logical names).  I linked the source in the issue where source is accessing these resources.

**Comment by elachlan** [comment]
@dreddy-work please see [EmbeddedResourceTests.cs](https://github.com/dotnet/winforms/pull/8358/files#diff-1cdfcb10896bac27708e37769d19b95b416df1d15512f98ba9a4935e07d5728a)

I have written tests to cover each resource added.

**Comment by dreddy-work** [comment]
`DummyNodeImage.ico` wasn't embedded into this proj file before but would do now. I do not see any references to this image anywhere. We should remove it after confirming.

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/EmbeddedResourceTests.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        [InlineData("System.Windows.Forms.Design.BindingFormattingDialog.Arrow")]
+        public void EmbeddedResource_ResourcesExist_Icon(string resourceName)
+        {
+            using Stream stream = assembly.GetManifestResourceStream(resourceName);
+            Assert.NotNull(stream);
+            Assert.NotNull(new Icon(stream));
+        }
+
+        [Theory]
+        [InlineData("System.Windows.Forms.Design.Behavior.MoverGlyph")]
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/EmbeddedResourceTests.cs:71]: We should have Ico for this. It seems we forgot to replace. Do you mind rebasing this and remove this special case afetr PR https://github.com/dotnet/winforms/pull/8370

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/EmbeddedResourceTests.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        [InlineData("System.Windows.Forms.Design.DataGridViewColumnsDialog.moveDown")]
+        [InlineData("System.Windows.Forms.Design.DataGridViewColumnsDialog.selectedColumns")]
+        [InlineData("System.Windows.Forms.Design.BindingFormattingDialog.Bound")]
+        [InlineData("System.Windows.Forms.Design.BindingFormattingDialog.Unbound")]
+        [InlineData("System.Windows.Forms.Design.BindingFormattingDialog.Arrow")]
+        public void EmbeddedResource_ResourcesExist_Icon(string resourceName)
+        {
+            using Stream stream = assembly.GetManifestResourceStream(resourceName);
+            Assert.NotNull(stream);
+            Assert.NotNull(new Icon(stream));
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/EmbeddedResourceTests.cs:67]: Please ensure that this icon is disposed.

**Comment by Tanya-Solyanik** [comment]
@elachlan  - please feel free to grab the ico from @dreddy-work PR and replace it in a dedicated commit.

**Comment by elachlan** [comment]
@Tanya-Solyanik Done!

**Comment by Tanya-Solyanik** [comment]
```
>DummyNodeImage.ico wasn't embedded into this proj file before but would do now. I do not see any references to this image anywhere. We should remove it after confirming.

This is where its supposed to be used apparently, according to the comment.
```

I wouldn't delete it.

**Comment by Tanya-Solyanik** [comment]
@elachlan  - could you please rebase this PR so that it has 2 commits:
1. replace the MoverGlyph with an ico
2. globs and tests

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/EmbeddedResourceTests.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+using System.Drawing;
+using System.Reflection;
+using Xunit;
+
+namespace System.Windows.Forms.Design.Tests
+{
+    public class EmbeddedResourceTests : IClassFixture<ThreadExceptionFixture>
+    {
+        // Get System.Windows.Forms.Design assembly.
+        // The type is irrelevant, its for the Icon constructor.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/EmbeddedResourceTests.cs:14]: I find this comment not clear.
Consider:
```
Get System.Windows.Forms.Design assembly to verify that it contains all the icons that the code uses.
```

Comment (elachlan) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/EmbeddedResourceTests.cs]: Done.

**Review by Tanya-Solyanik — APPROVED** [review]

**Comment by elachlan** [comment]
Unrelated test failure.

**Comment by Tanya-Solyanik** [comment]
System.Windows.Forms.Tests.CursorConverterTests.CursorConverter_ConvertTo_UnknownToInstanceDescriptor_ThrowsNotSupportedException

**Error message**
```
Assert.Throws() Failure\r\nExpected: typeof(System.NotSupportedException)\r\nActual:   typeof(System.Reflection.TargetInvocationException): Exception has been thrown by the target of an invocation.\r\n---- System.Reflection.TargetInvocationException : Exception has been thrown by the target of an invocation.\r\n-------- System.ArgumentException : Image format is not valid. The image file may be corrupted. (Parameter 'stream')\r\n------------ System.Runtime.InteropServices.COMException : Error HRESULT E_FAIL has been returned from a call to a COM component.
```

**Stack trace**
```
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index)
   at System.Windows.Forms.CursorConverter.ConvertTo(ITypeDescriptorContext context, CultureInfo culture, Object value, Type destinationType) in /_/src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs:line 129
   at System.ComponentModel.TypeConverter.ConvertTo(Object value, Type destinationType)
   at System.Windows.Forms.Tests.CursorConverterTests.<>c__DisplayClass10_0.<CursorConverter_ConvertTo_UnknownToInstanceDescriptor_ThrowsNotSupportedException>b__0() in /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorConverterTests.cs:line 117
----- Inner Stack Trace -----
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index)
   at System.Windows.Forms.CursorConverter.ConvertTo(ITypeDescriptorContext context, CultureInfo culture, Object value, Type destinationType) in /_/src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs:line 129
   at System.ComponentModel.TypeConverter.ConvertTo(Object value, Type destinationType)
   at System.Windows.Forms.Tests.CursorConverterTests.<>c__DisplayClass10_0.<CursorConverter_ConvertTo_UnknownToInstanceDescriptor_ThrowsNotSupportedException>b__0() in /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorConverterTests.cs:line 117
----- Inner Stack Trace -----
   at System.Windows.Forms.Cursor.LoadPicture(Interface stream, String paramName) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs:line 437
   at System.Windows.Forms.Cursor..ctor(Stream stream) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs:line 90
   at System.Windows.Forms.Cursor..ctor(Type type, String resource) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs:line 73
   at System.Windows.Forms.Cursors.get_HSplit() in /_/src/System.Windows.Forms/src/System/Windows/Forms/Cursors.cs:line 69
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
----- Inner Stack Trace -----
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHR(Int32 errorCode, IntPtr errorInfo)
   at Windows.Win32.Foundation.HRESULT.ThrowOnFailure(IntPtr errorInfo) in /_/src/System.Windows.Forms.Primitives/src/Microsoft.Windows.CsWin32/Microsoft.Windows.CsWin32.SourceGenerator/Windows.Win32.HRESULT.g.cs:line 63
   at Windows.Win32.System.Com.IPersistStream.Load(IStream* pStm) in /_/src/System.Windows.Forms.Primitives/src/Microsoft.Windows.CsWin32/Microsoft.Windows.CsWin32.SourceGenerator/Windows.Win32.IPersistStream.g.cs:line 156
   at System.Windows.Forms.Cursor.LoadPicture(Interface stream, String paramName) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs:line 410
```

**Comment by elachlan** [comment]
That test failure could be related to #8367

**Comment by elachlan** [comment]
Thanks @Tanya-Solyanik! I will get started on `System.Windows.Forms` soon.

**Comment by Tanya-Solyanik** [comment]
Oops, I shouldn't had marked it as automerge, it was supposed to be merged as 2 commits. Sorry, @elachlan for making you rebase it and then not using it...

**Pull request merged**